addons/website_event_sale/controllers/main.py

```diff
@@ -70,3 +71,15 @@ def registration_confirm(self, event, **post):
                 request.website.sale_reset()
 
         return res
+
+
+class PaymentPortalOnsite(PaymentPortal):
+
+    def _validate_transaction_for_order(self, transaction, sale_order_id):
+        """
+        Throws a ValidationError if the user tries to pay for a ticket which isn't available
+        """
+        super()._validate_transaction_for_order(transaction, sale_order_id)
+        sale_order = request.env['sale.order'].browse(sale_order_id).exists().sudo()
+        tickets = sale_order.order_line.filtered(lambda line: line.product_type == 'event').mapped('event_ticket_id')
```

Comment:
Would specify `sale_order_sudo`, in case this override suddenly gets bigger one day

And fyi you can just `.event_ticket_id`, no need to map over relational fields it's the same result